internal/pkg/platform/model_test/create_test.go

```diff
@@ -0,0 +1,228 @@
+package model_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/keboola/go-client/pkg/keboola"
+	"github.com/keboola/go-utils/pkg/orderedmap"
+	"github.com/stretchr/testify/assert"
+
+	_ "github.com/mattn/go-sqlite3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/platform/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/platform/model/enttest"
+	"github.com/keboola/keboola-as-code/internal/pkg/platform/model/key"
+)
+
+func TestCreate_SetParent(t *testing.T) {
+	t.Parallel()
+	ctx, client, tx := txForTest(t)
+
+	branch, err := branchBuilder(tx).
+		SetBranchID(123).
+		Save(ctx)
+	assert.NoError(t, err)
+
+	config, err := configBuilder(tx).
+		SetParent(branch).
+		SetComponentID("keboola.my-component").
+		SetConfigID("my-config").
+		Save(ctx)
+	assert.NoError(t, err)
+
+	row, err := rowBuilder(tx).
+		SetParent(config).
+		SetRowID("my-row").
+		Save(ctx)
+	assert.NoError(t, err)
+
+	assert.NoError(t, tx.Commit())
+	checkResults(t, ctx, client, branch, config, row)
```

Comment:
`entgo` vygeneruje `create` a `update` builder pre kazdu entitu.

Key je immutable a da sa nastavit iba pre `create` (pre `uptate` tam dane setters nie su).

Reply:
Primarny kluc je teda zlozeny, a da sa nastavit roznymi sposobmi.

`TestCreate_*` testy testuju, ze tieto sposoby su rovnocenne.

Tym sa testuje, ze kod vygenerovany cez `primarykey` extension pracuje spravne.

------

Tieto testy sa uz nebudu rozsirovat.
Ak to funguje spravne pre tieto entity, tak to bude fungovat pre vsetky.

Reply:
Iba pripomenie, ze vysledkom tejto create operacie su tabulky v `in-memory sqlite`.

`100 000` operacii trva `~7s`, takze s rychlostou by nemal byt problem.